### PR TITLE
Use testcontainers for test, instead of local zerotier-one

### PR DESCRIPTION
**Pull request by laduke**
Before, we assumed that the host/dev-env had a zerotier-one service running.

Now we can start a container and use the zerotier-one in there, and then throw it away.
No messing up your local service. No having to switch its version around. Should work for anyone with a docker runtime.

### CODE REVIEW

File: test/integration.test.ts [hunk 1]
```diff
@@ -13,18 +14,16 @@ import { AnyValidateFunction } from "ajv/dist/core.js";
...
 const addFormats = _addFormats.default;
 
-if (!process.env.AUTH_TOKEN) {
-  console.error("No AUTH_TOKEN env var set. Exiting.");
-  process.exit(1);
-}
+const ZT_VERSION = process.env.ZT_VERSION || "1.12.2"
 
-const isNextApiVersion = process.env.ENABLE_UNSTABLE;
+function compareVersions(a: string, b: string) {
```
Comment (rcoder) [inline test/integration.test.ts]: The expected usage of this function is a little confusing to me. (I.e., is it for sorting? Exact version matches? Etc.) The fact that `localeCompare` does _roughly_ the right thing for some version strings seems a little shaky, esp. given that this runs in the local environment's default locale.

I also don't think it does exactly what we want in even some pretty natural cases:

```
// good!
> "1.15.0".localeCompare("1.14.0")
1
// less good
> "1.15.0".localeCompare("1.2.9")
-1
```

Might be less fiddly to just use the [semver](https://www.npmjs.com/package/semver) library to do explicit version matching instead.

File: package.json [hunk 2]
```diff
@@ -50,6 +50,7 @@
     "typescript": "^5.3.3"
   },
   "devDependencies": {
-    "borp": "^0.9.1"
+    "borp": "^0.9.1",
```
Comment (rcoder) [inline package.json:53]: Is this actually used?

File: test/integration.test.ts [hunk 3]
```diff
@@ -59,32 +58,51 @@ function assertValid(
...
+
+  before(async () => {
+    container = await new GenericContainer(`zerotier/zerotier:${ZT_VERSION}`)
+      .withExposedPorts(9993)
+      .withPrivilegedMode()
+      .withAddedCapabilities("NET_ADMIN")
+      .withEnvironment({ ZEROTIER_API_SECRET })
+      .withCopyContentToContainer([
+        {
+          content: `{ "settings": { "allowManagementFrom": ["0.0.0.0/0"] } }`,
```
Comment (rcoder) [inline test/integration.test.ts:75]: Nitpick: if we use JSON.stringify on a native JS object for injecting this kind of configuration (vs. string templates) we could pretty easily make the specific config you pass overridable/composable.

E.g.:

```
content: JSON.stringify({ ...DEFAULT_LOCAL_CONF, allowManagementFrom: [ "0.0.0.0/0" ]})
```

(Again, just a stylistic thing, and totally fine to ignore until we actually have >1 config file body to populate + inject.)

**Review by rcoder — COMMENTED** [review]
This generally looks pretty good. I'm not totally confident in the current `compareVersions` implementation and made a suggestion there, but nothing else is a blocker.

File: package.json (hunk 2 above)
Comment (laduke) [reply package.json:53]: it's the test runner :)

File: test/integration.test.ts (hunk 1 above)
Comment (laduke) [reply test/integration.test.ts]: i want to delete this entirely

File: test/integration.test.ts (hunk 3 above)
Comment (laduke) [reply test/integration.test.ts:75]: just a hack to get it to work as expected in docker for mac

File: test/integration.test.ts (hunk 1 above)
Comment (laduke) [reply test/integration.test.ts]: i went back and forth on the library. i'll add the library

Comment (laduke) [reply test/integration.test.ts]: OK that's added. 

by the way 
```
"1.15.0".localeCompare("1.14.0", undefined, { numeric: true, sensitivity: 'base' })
1
"1.15.0".localeCompare("1.2.9", undefined, { numeric: true, sensitivity: 'base' })
1 
```

**Review by rcoder — APPROVED** [review]
LGTM

**Pull request merged**